Sources/StreamChat/Controllers/MessageController/MessageController_Tests.swift

```diff
@@ -386,12 +386,16 @@ final class MessageController_Tests: StressTestCase {
         try client.databaseContainer.writeSynchronously { session in
             replyDTO = try session.saveMessage(payload: reply, for: self.cid)
         }
+        
+        let context = try XCTUnwrap(replyDTO?.managedObjectContext)
     
-        // Assert `insert` entity change is received by the delegate
-        AssertAsync.willBeEqual(
-            delegate.didChangeReplies_changes,
-            [.insert((replyDTO?.asModel())!, index: [0, 0])]
-        )
+        context.performAndWait {
+            // Assert `insert` entity change is received by the delegate
+            AssertAsync.willBeEqual(
+                delegate.didChangeReplies_changes,
+                [.insert((replyDTO?.asModel())!, index: [0, 0])]
+            )
+        }
```

Comment:
Doing the asserts inside `performAndWait` is a little bit odd. Since we only compare the model object here, anyway, can't we just get it from the `viewContext`?

Reply:
That could be an option or maybe just inside `performAndWait` I can convert `replyDTO` to model and then compare it on main thread?

Reply:
@olejnjak I believe both should work 👍